packages/sdk/src/services/item-service.ts

```diff
@@ -121,103 +129,136 @@ export class ItemService {
     });
   }
 
-  public async update(vaultAccessToken: string, dek: EncryptionKey, config: ItemUpdateData) {
+  public async update(credentials: IVaultToken & IDEK, config: ItemUpdateData) {
     const slots_attributes = await Promise.all(
-      (config.slots || []).map(slot => this.encryptSlot(slot, dek))
+      (config.slots || []).map(slot =>
+        ItemService.encryptSlot(slot, credentials.data_encryption_key)
+      )
     );
 
-    return await this.vaultAPIFactory(vaultAccessToken).ItemApi.itemsIdPut(config.id, {
+    return await this.getAPI(credentials.vault_access_token).itemsIdPut(config.id, {
       item: {
         label: config.label,
         slots_attributes,
       },
     });
   }
 
-  public async removeSlot(slotId: string, vaultAccessToken: string) {
+  public async removeSlot(credentials: IVaultToken, slotId: string) {
     this.logger.log('Removing slot');
-    await this.vaultAPIFactory(vaultAccessToken).SlotApi.slotsIdDelete(slotId);
+    await this.vaultAPIFactory(credentials.vault_access_token).SlotApi.slotsIdDelete(slotId);
```

Comment:
Unless I'm just getting confused mixing `this.vaultAPIFactory` and `this.getAPI` feels a bit confusing to me?

Reply:
I made vault/keystoreAPIFactory protected in all services, but perhaps it's confusing mixing getAPI and APIFactory methods inside the class? Would it be less confusing if I only used APIFactory methods within the class say?

Reply:
Yeah I think I get that `getApi` is like a specific api (e.g. `SlotsApi`) but it's also a bit confusing to read through having both in there.